renderdoc/driver/gl/wrappers/gl_texture_funcs.cpp

```diff
@@ -1388,55 +1393,121 @@ void WrappedOpenGL::glCopyImageSubData(GLuint srcName, GLenum srcTarget, GLint s
                                  dstData.curType != eGL_TEXTURE_CUBE_MAP_ARRAY)
                                     ? RDCMAX(1, dstData.depth >> dstLevel)
                                     : dstData.depth;
-        if(srcX == 0 && srcY == 0 && srcZ == 0 && dstX == 0 && dstY == 0 && dstZ == 0 &&
-           srcLevelWidth == dstLevelWidth && srcLevelHeight == dstLevelHeight &&
-           srcLevelDepth == dstLevelDepth)
+
+        bytebuf srcCd;
+        // if we have source compressed data to copy
+        if(srcData.compressedData.find(srcLevel) != srcData.compressedData.end())
         {
-          dstData.compressedData[dstLevel] = srcData.compressedData[srcLevel];
+          // TODO: avoid copy
+          srcCd = srcData.compressedData[srcLevel];
         }
-        else
+        else if(!srcIsCompressed &&
+                (srcData.curType == eGL_TEXTURE_2D || srcData.curType == eGL_TEXTURE_2D_ARRAY))
```

Comment:
Is it worth adding an error message if the source texture is not 2D or 2D array? it could be a cubemap or 3D texture as well, copying a single slice.

Reply:
I'm a bit unsure about supporting 3D or Cubemap here (just remove line 1405 and add corresponding `eGL_TEXTURE_BINDING_xxx`), which is not tested under my usages.

Reply:
Yes that's totally fine to not support it, I just mean to add an error message for that case. It looks like at the moment the call will be silently dropped.